### PR TITLE
Fix using non-square images with UNet2DModel and DDIM/DDPM pipelines

### LINKED ISSUE #1262: train a diffusion model on rectangle images
### Describe the bug

How to train the pictures with Rectangle ,I tried 128*128, and it is fine,but 128*256 input then there are error of tuple size, does anyone know how to solve it?
I just use the diffusers_training_example on colab
image

What I change:
1. 
@dataclass
class TrainingConfig:
    # image_size = 128  # the generated image resolution
    image_size_h = 128  # my change
    image_size_w = 256  # my change

2. 
preprocess = transforms.Compose(
    [
        transforms.Resize((config.image_size_h, config.image_size_w)),
        transforms.RandomHorizontalFlip(),

3. 
model = UNet2DModel(
    sample_size=(config.image_size_h,config.image_size_w)

Here is the error
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-30-4f9138074009> in <module>
      2 args = (config, model, noise_scheduler, optimizer, train_dataloader, lr_scheduler)
      3 
----> 4 notebook_launcher(train_loop, args, num_processes=1)

4 frames
/usr/local/lib/python3.7/dist-packages/diffusers/pipelines/ddpm/pipeline_ddpm.py in __call__(self, batch_size, generator, output_type, return_dict, **kwargs)
     69         image = torch.randn(
     70             (batch_size, self.unet.in_channels, self.unet.sample_size_w, self.unet.sample_size_h),
---> 71             generator=generator,
     72         )
     73         image = image.to(self.device)

TypeError: randn(): argument 'size' must be tuple of ints, but found element of type tuple at pos 3

### Reproduction

_No response_

### Logs

_No response_

### System Info

the example on colab

### PR DESCRIPTION
**Pull request by jenkspt**
Minimal changes to fix #1262.

- Corrects `UNet2DModel` docstring for parameter `sample_size`
- Adds check in DDPM/DDIM pipelines to handle tuple or int value for `sample_size` attribute of `UNet2DModel`

I still need to add tests for non square images w/ `UNet2DModel`. This is my first pull request -- I'm not to familiar with the testing yet. I'm open to suggestions on a good way to parameterize `sample_size` in `tests/test_pipelines.py`.

Also `UNet2DConditional` should also be updated ~~(as well as flax versions)~~. Not sure if this should be included in this pull request.

### CODE REVIEW

File: src/diffusers/pipelines/ddim/pipeline_ddim.py [hunk 1]
```diff
@@ -89,7 +89,11 @@ def __call__(
             generator = None
 
         # Sample gaussian noise to begin loop
-        image_shape = (batch_size, self.unet.in_channels, self.unet.sample_size, self.unet.sample_size)
+        if isinstance(self.unet.sample_size, int):
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/ddim/pipeline_ddim.py:92]: Isn't this the same?

**Review by patrickvonplaten — COMMENTED** [review]
Ok for me - thanks for fixing this :-) 
Let me know when the PR is not WIP anymore.

**Comment by jenkspt** [comment]
Added parameterized test case `test_uncond_unet_components` to `PipelineFastTests`.
In order to support this I made a few changes:
- renamed `test_components` to `test_stable_diffusion_components` to disambiguate
- removed `property` decorators from all `dummy` prefixed methods so that they can be parameterized

**Comment by jenkspt** [comment]
Also looks like code quality checks are failing due to recent commits to main

**Review by patil-suraj — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**